src/group/libp2p_group_gossip_server.erl

```diff
@@ -147,6 +147,18 @@ handle_cast({add_handler, Key, Handler}, State=#state{handlers=Handlers}) ->
     {noreply, State#state{handlers=maps:put(Key, Handler, Handlers)}};
 handle_cast({request_target, inbound, WorkerPid, _Ref}, State=#state{}) ->
     {noreply, stop_inbound_worker(WorkerPid, State)};
+handle_cast({clear_target, _Kind, _WorkerPid, Ref}, State=#state{workers = Workers}) ->
+    lager:debug("clearing target for worker ~p ", [_WorkerPid]),
+    %% the ref is stable across restarts, so use that as the lookup key
+    case lookup_worker(Ref, #worker.ref, State) of
+        Worker=#worker{} ->
```

Comment:
In theory the workerpid in here should match the one passed in the message?

Reply:
yes, Worker.pid() should equal WorkerPid from the message.  As per the comment, the ref is used for the lookup as that is consistent across restarts of the worker